api/src/wfl/module/covid.clj

```diff
@@ -25,8 +18,65 @@
 (defn create-covid-workload!
   [tx {:keys [source sink executor] :as request}])
 
-(defn start-covid-workload!
-  [])
+;; TODO: move this to the queue-based skeleton later
+(defn ^:private snapshot-created?
+  "Check if `snapshot-job-id` has done in TDR,
+   return the resulting snapshot id if so."
+  [snapshot-job-id]
+  (some->> snapshot-job-id
+           (datarepo/job-done?)
+           :id))
+
+(defn ^:private go-create-snapshot!
+  "Create snapshot in TDR from `dataset`, `table`
+   and `row-ids` and write job info as well as rows
+   into `source-details-name` table."
+  [tx dataset table source-details-name [start end] row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
+        job-id (-> (datarepo/make-snapshot-request dataset columns table row-ids)
+                   (update :name util/randomize)
```

Comment:
I don't like the idea of using randomisation in product code. Furthermore, this name will mean nothing to users of the workspace.
Can we work along the lines of [Dataset Name][YYYYMMDDTHHMMSS] - the slightly unconventional timestamp is because it has to be an identifier.

Reply:
Note that timestamp is when the snapshot was created

Reply:
> the slightly unconventional timestamp is because it has to be an identifier

Can you elaborate on this, are there forbidden characters to be aware of? I ask because I should probably make a similar change to the snapshot reference naming.  Otherwise I'd just cast `now` to string and get something like `2021-05-07T14:47:54.440594-04:00`.

Reply:
I updated the `(util/utc-now)` so it optionally takes a `format`  argument now, e.g.

```clojure
(util/utc-now "YYYYMMDD'T'HHMMSS")
=> "202105130T190537"
```